superset/db_engine_specs/presto.py

```diff
@@ -944,22 +955,22 @@ def where_latest_partition(cls, table_name, schema, database, qry, columns=None)
         column_names = {column.get("name") for column in columns or []}
         for col_name, value in zip(col_names, values):
             if col_name in column_names:
-                qry = qry.where(Column(col_name) == value)
-        return qry
+                query = query.where(Column(col_name) == value)
+        return query
 
     @classmethod
     def _latest_partition_from_df(cls, df):
         if not df.empty:
             return df.to_records(index=False)[0].item()
 
     @classmethod
-    def latest_partition(cls, table_name, schema, database, show_first=False):
+    def latest_partition(
+        cls, table_name: str, schema: str, database, show_first: bool = False
+    ) -> Tuple[str, str]:
```

Comment:
This typing isn't quite right here:

I believe we return a `Tuple[List[str], Optional[List[str]]]`

`column_names` is a list (as we call `len` on it) and the second piece of the tuple can be `None` (see method above)

Reply:
My bad; I got lazy and just read the example in the docstring and assumed it was still up to date. Thanks for managing to get this far in this review without falling asleep 😃